azure-iot-device/azure/iot/device/iothub/transport/mqtt/mqtt_transport.py

```diff
@@ -329,84 +323,32 @@ def _on_provider_disconnect_complete(self):
             self._disconnect_callback = None
             callback()
 
-    def _on_provider_publish_complete(self, mid):
-        """
-        Callback that is called by the provider when it receives a PUBACK from the service
-
-        :param mid: message id that was returned by the provider when `publish` was called.  This is used to tie the
-            PUBLISH to the PUBACK.
-        """
-        if mid in self._in_progress_actions:
-            callback = self._in_progress_actions[mid]
-            del self._in_progress_actions[mid]
-            callback()
-        else:
-            logger.warning("PUBACK received with unknown MID: %s", str(mid))
-            self._responses_with_unknown_mid[
-                mid
-            ] = mid  # storing MID for now.  will probably store result code later.
-
-    def _on_provider_subscribe_complete(self, mid):
-        """
-        Callback that is called by the provider when it receives a SUBACK from the service
-
-        :param mid: message id that was returned by the provider when `subscribe` was called.  This is used to tie the
-            SUBSCRIBE to the SUBACK.
-        """
-        if mid in self._in_progress_actions:
-            callback = self._in_progress_actions[mid]
-            del self._in_progress_actions[mid]
-            callback()
-        else:
-            logger.warning("SUBACK received with unknown MID: %s", str(mid))
-            self._responses_with_unknown_mid[
-                mid
-            ] = mid  # storing MID for now.  will probably store result code later.
-
     def _on_provider_message_received_callback(self, topic, payload):
         """
         Callback that is called by the provider when a message is received.  This message can be any MQTT message,
         including, but not limited to, a C2D message, an input message, a TWIN patch, a twin response (/res), and
         a method invocation.  This function needs to decide what kind of message it is based on the topic name and
         take the correct action.
 
-        :param topic: MQTT topic name that the message arrived on
-        :param payload: Payload of the message
+        :param str topic: MQTT topic name that the message arrived on
+        :param bytes payload: Payload of the message as a bytestring
         """
-        logger.info("Message received on topic %s", topic)
+        logger.info("Message received on topic {}".format(topic))
         message_received = Message(payload)
         # TODO : Discuss everything in bytes , need to be changed, specially the topic
```

Comment:
># TODO : Discuss everything in bytes , need to be changed, specially the topic [](start = 8, length = 78)

u can remove this TODO. I kept it because of the `_topic` mistaken use.